### PR TITLE
Version 2.5.6

### PR DESCRIPTION
**Pull request by SC426**
conda-lock 2.5.6

**Destination channel:** defaults

### Links

- [PKG-3620](https://anaconda.atlassian.net/browse/PKG-3620)
- [Upstream repository](https://github.com/conda/conda-lock/tree/v2.5.6)
- [Upstream changelog/diff](https://github.com/conda/conda-lock/releases)
- Relevant dependency PRs:
  - [PKG-4349](https://anaconda.atlassian.net/browse/PKG-4349)

### Explanation of changes:

- Update version and dependencies

[PKG-3620]: https://anaconda.atlassian.net/browse/PKG-3620?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PKG-4349]: https://anaconda.atlassian.net/browse/PKG-4349?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,54 +1,53 @@
...
 
 requirements:
   host:
     - python
-    - hatchling
-    - hatch-requirements-txt
-    - hatch-vcs
     - pip
     - setuptools
     - wheel
```
Comment (Jrice1317) [inline recipe/meta.yaml]:
```suggestion
```

Comment (Jrice1317) [reply recipe/meta.yaml]: Since hatchling is used, setuptools and wheel are not needed.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,54 +1,53 @@
...
+    - hatchling
+    - hatch-vcs
   run:
     - python
-    - cachecontrol-with-filecache >=0.12.9
-    - cachy >=0.3.0
     - click >=8.0
```
Comment (Jrice1317) [inline recipe/meta.yaml:26]: Not a blocker, but maybe adding some of the same comments and/or listing poetry 1.1.15 explicitly would be nice. https://github.com/conda/conda-lock/blob/v2.5.6/pyproject.toml#L44

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,54 +1,53 @@
...
 source:
   url: https://pypi.io/packages/source/c/conda-lock/conda_lock-{{ version }}.tar.gz
-  sha256: 55c115a852e63395db8a6a90cf2892aab0543959f2d0bafb83158b6863831b8a
+  sha256: 323db8b6438b46648c4ba4dab81547851ee8266dbe913680f7837c94cc5f1023
 
 build:
   number: 0
-  skip: True  # [py<36]
-  script: {{ PYTHON }} -m pip install . -vv --no-deps
+  skip: True  # [py<38]
```
Comment (Jrice1317) [inline recipe/meta.yaml:13]: Also not a blocker, but it seems `s390x` can be skipped as it's not listed in the [platforms](https://github.com/conda/conda-lock/blob/v2.5.6/pyproject.toml#L156).

**Review by danpetry — APPROVED** [review]
subject to jaida's comments being addressed

File: recipe/meta.yaml (hunk 3 above)
Comment (cbouss) [reply recipe/meta.yaml:13]: We should provide s390x if we can. I don't this section applying to us.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:13]: Umm, I don't think these platforms are necessarily what's supported. We should keep s390x.

File: recipe/meta.yaml (hunk 1 above)
Comment (SC426) [reply recipe/meta.yaml]: Fixed!

**Review by Jrice1317 — APPROVED** [review]

**Pull request merged**